### PR TITLE
Add new interface for Client Error and a timeout to 'get Artifact Checksum'

### PR DESCRIPTION
**Pull request by Or-Geva**

### CODE REVIEW

File: src/HttpClient.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export class HttpClient {
     private static readonly AUTHORIZATION_HEADER: string = 'Authorization';
     private static readonly USER_AGENT_HEADER: string = 'User-Agent';
     private static readonly DEFAULT_RETRIES: number = 3;
+    public static readonly DEFAULT_TIMEOUT: number = 2000;
```
Comment (yahavi) [inline src/HttpClient.ts:9]: Default timeout in what? Seconds? Millisecond? Nano?

File: test/tests/Artifactory/ArtifactoryDownloadClient.spec.ts [hunk 2]
```diff
@@ -70,4 +71,14 @@ describe('Artifactory Download tests', () => {
         expect(checksum.sha1).toBeTruthy();
         expect(checksum.sha256).toBeTruthy();
     });
+
+    test('Build artifact download checksum test', async () => {
```
Comment (yahavi) [inline test/tests/Artifactory/ArtifactoryDownloadClient.spec.ts:75]:
```suggestion
    test('Build artifact download checksum timeout test', async () => {
```

File: test/tests/Artifactory/ArtifactoryDownloadClient.spec.ts [hunk 3]
```diff
@@ -70,4 +71,14 @@ describe('Artifactory Download tests', () => {
...
+        const dummyClient: JfrogClient = (jfrogClient = new JfrogClient({
+            artifactoryUrl: 'http://localhost:8080/artifactory',
```
Comment (yahavi) [inline test/tests/Artifactory/ArtifactoryDownloadClient.spec.ts:77]:
```suggestion
            artifactoryUrl: 'http://localhost:8080/artifactory',
            logger: TestUtils.createTestLogger(),
```

Prevents the ugly log message in the middle of the tests:
image

**Review by yahavi — CHANGES_REQUESTED** [review]

File: src/HttpClient.ts (hunk 1 above)
Comment (Or-Geva) [reply src/HttpClient.ts:9]: Millisecond

**Pull request merged**